src/structure/real.rs

```diff
@@ -0,0 +1,267 @@
+use num::{Float, Num, FromPrimitive};
+use std::ops::{Neg, AddAssign, MulAssign, SubAssign, DivAssign};
+use cmp::ApproxEq;
+use structure::FieldApprox;
+
+#[allow(missing_docs)]
+
+/// Trait shared by all approximate reals.
+///
+/// Approximate reals are approximate fields aquipped with funtinos that are commonly used on
+/// reals. The results of those functions only have to be approximately equal to the actual
+/// theoretical values.
+pub trait RealApprox: Copy + Num + FieldApprox + FromPrimitive +
+                      Neg<Output = Self> + AddAssign + MulAssign + SubAssign + DivAssign +
+                      ApproxEq<Eps = Self> + Ord {
```

Comment:
I'm not sure I understand what you mean. How could we use a where clause in this context?